bookkeeper-common/src/test/java/org/apache/bookkeeper/common/util/CronExpressionTest.java

```diff
@@ -0,0 +1,848 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.bookkeeper.common.util;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import java.time.LocalDate;
+import java.time.YearMonth;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TimeZone;
+import org.apache.bookkeeper.common.util.CronExpression.CronFieldType;
+import org.apache.bookkeeper.common.util.CronExpression.DayOfMonthField;
+import org.apache.bookkeeper.common.util.CronExpression.DayOfWeekField;
+import org.apache.bookkeeper.common.util.CronExpression.SimpleField;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * A test for CronExpression.
```

Comment:
Add a comment this file is based on https://github.com/frode-carlsen/cron/blob/master/java8/src/test/java/fc/cron/CronExpressionTest.java